arealite/engine/sglang_local.py

```
@@ -0,0 +1,378 @@
+
+    async def agenerate(self, req: LLMRequest) -> LLMResponse:
+        """Async version of generate using local sglang engine."""
+        if not hasattr(self, "engine") or self.engine is None:
+            raise RuntimeError("Local SGLang engine is not initialized, cannot generate.")
+            
+        # Prepare request payload
+        gconfig = req.gconfig
+        stop_token_ids = gconfig.stop_token_ids
+
+        if gconfig.n_samples != 1:
+            raise ValueError(
+                "LocalSGLangEngine does not support n_samples > 1. "
+                "Please call generate for multiple times with n_samples = 1."
+            )
+        sample_params = {
+            "top_p": gconfig.top_p,
+            "top_k": gconfig.top_k,
+            "max_new_tokens": gconfig.max_new_tokens,
+            "temperature": 0.0 if gconfig.greedy else gconfig.temperature,
+            "stop_token_ids": stop_token_ids,
+        }
+
+        completions = ""
+        prompt = req.text if req.text else None
+        input_ids = req.input_ids if req.input_ids else None
+
+        # Make request
+        start_time = time.perf_counter()
+        accumulated_output_tokens = []
+        accumulated_output_logprobs = []
+        accumulated_versions = []
+        stop_reason = "length"
+        while (
+            stop_reason != "stop"
+            and len(accumulated_output_tokens) < gconfig.max_new_tokens
+        ):
+
+            try:
+                outputs = await self.engine.async_generate(
+                    prompt=prompt,
+                    input_ids=input_ids,
+                    sampling_params=sample_params,
+                    return_logprob=True
+                )
+
+                completions += outputs.get("text", "")
+                if prompt is None:
+                    prompt = outputs.get("text", "")
+                else:
+                    prompt += outputs.get("text", "")
+                
+                if "meta_info" in outputs:
```

Can you make a note about why this condition will be true?

---

I mean the code should directly raise an error if such a key is absent.